### ISSUE TITLE
Help please

### ISSUE THREAD

**Issue by heon01px2020** [cedrickchee/pytorch-android#1]
In this project, it seem to be using resnet or squeezenet. However, in a project I'm doing of my own, I am currently using mobilenetv2. Currently, I have converted the mobilenetv2 model from pytorch to caffe2. If I use this caffe2 file, would it work seamlessly after replacing the _init_net.pb and _predict_net.pb or do I need to make other modifications? Thank you.

**Comment by cedrickchee** [comment]
> would it work seamlessly after replacing the _init_net.pb and _predict_net.pb or do I need to make other modifications?

It should. MobileNet v2 (or SqueezeNet) input (data)/output (prediction) tensor dimension is the same. So, it work seamlessly (no code changes required to the `native-lib.cpp`).

Unfortunately, when I tested with a quantized MobileNet v2 today, the Android app crashed due to Caffe2 quantized fc op:

![mobilenet_v2_caffe2_predictnet_crash](https://user-images.githubusercontent.com/145605/50679588-02f36380-103f-11e9-86a6-dc33cdd03f81.png)

```
2019-01-04 16:27:24.093 32484-32545/facebook.f8demo A/native: [F int8_fc_op.h:43] Check failed: K == W.t.size(1) 
2019-01-04 16:27:24.093 32484-32545/facebook.f8demo A/native: terminating.
    
    
    --------- beginning of crash
2019-01-04 16:27:24.093 32484-32545/facebook.f8demo A/libc: Fatal signal 6 (SIGABRT), code -6 in tid 32545 (Camera Backgrou), pid 32484 (facebook.f8demo)
```

I have not tested with non-quantized MobileNet v2 yet.

**Comment by heon01px2020** [comment]
Ok, thank you so much! I had another question which was that I have an image classifier model, not an object detection model. Would there be any modifications needed?

**Comment by heon01px2020** [comment]
Yes, the Mobilenet v2 model I made is a classification model. I thought that your tutorial used object detection. Does that require me to make any modifications to the code? What I want to do is have the app take frames from the video, classify the image, and output the classification on the screen. Thank you!

**Comment by cedrickchee** [comment]
My tutorials used image classification models. It _should_ not require any modifications to the code if you put aside and do not count the previous reply with the MobileNet v2 error. I can't confirm at this point if this problem is due to Caffe2 or my own code, but from the error messages, I can tell the problem is in Caffe2.

> What I want to do is have the app take frames from the video, classify the image, and output the classification on the screen.

The Android app is doing exactly that. Try it and find out yourself! :slightly_smiling_face:

**Comment by heon01px2020** [comment]
Ok, thank you! I tried again without my model, but it couldn't build because I am missing jniLibs.armeabi-v7a/libcaffe2.a Do you know how to get it?

**Comment by cedrickchee** [comment]
The steps to download libcaffe2.a and other library files are in the [README](https://github.com/cedrickchee/pytorch-android#documentation).

> If you prefer not to build PyTorch from master, I have committed the files for the Caffe2 bits into this git repo. Unfortunately, not all files are uploaded to GitHub due to file size limit.
> Grab the remaining large files from my [Google Drive](https://drive.google.com/drive/folders/1e1lFa8W2rkA8M7mhahe9VSiSMnl0a4Ab?usp=sharing.

I don't know how you missed it. May be I should have been clearer.
